src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -754,8 +754,7 @@ private void publishInternal(String subject, String replyTo, Headers headers, by
 
         NatsMessage nm = new NatsMessage(subject, replyTo, new Headers(headers), data, utf8mode);
 
-        Connection.Status stat = this.status;
-        if ((stat == Status.RECONNECTING || stat == Status.DISCONNECTED)
+        if ((this.status == Status.RECONNECTING || this.status == Status.DISCONNECTED)

```

Comment:
I think I see why the original author did that - to reduce issues around thread safety.  Longer term, the status checks need to be protected or atomic.  At the least, status should be volatile.

Reply:
reverted